connmgr.go

```diff
@@ -68,12 +70,25 @@ type peerInfo struct {
 // pruning those peers with the lowest scores first, as long as they are not within their
 // grace period.
 func (cm *BasicConnMgr) TrimOpenConns(ctx context.Context) {
```

Comment:
The fact that this returns early if there's an existing trim but waits otherwise is a bit funky. What about just *triggering* a trim and always going async?

Reply:
This method is always called as a goroutine, so I think we’re cobered? (Unless I misunderstood the remark)

Reply:
Well, that's the only place *we're* calling it but it's a public method. I'd just be happier if it behaved consistently.

Reply:
Gotcha!

Reply:
I'll add a TODO. Not sure if it's best to make the second, third, etc. trims wait until the first one is done, or if we can return an error. I feel the caller will want us to block until the first trim is done.